### PR TITLE
SAK-45025 use new AlphaNumericComparator in samigo, assignments, lessons

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/PublishedAssessmentSettingsBean.java [hunk 1]
```diff
@@ -1468,36 +1470,25 @@ public String editSettingBeforePublish() {
...
+          Collection<Group> groups = site.getGroups();
+          if (groups != null && groups.size() > 0) {
+              for (Group group : groups) {
+                  sortedSelectItems.put(group.getTitle(), new SelectItem(group.getId(), group.getTitle()));
+              }
+
+              groupSelectItems = sortedSelectItems.values().toArray(new SelectItem[0]);
+          }
+      } catch (IdUnusedException ex) {
+          // No site available
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/PublishedAssessmentSettingsBean.java:1488]: probably need a log.warn here

Comment (ottenhoff) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/PublishedAssessmentSettingsBean.java:1488]: ok coming up

Comment (ottenhoff) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/PublishedAssessmentSettingsBean.java]: resolved

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/PublishedAssessmentSettingsBean.java [hunk 2]
```diff
@@ -1468,36 +1470,25 @@ public String editSettingBeforePublish() {
...
+          Collection<Group> groups = site.getGroups();
+          if (groups != null && groups.size() > 0) {
+              for (Group group : groups) {
+                  sortedSelectItems.put(group.getTitle(), new SelectItem(group.getId(), group.getTitle()));
+              }
+
+              groupSelectItems = sortedSelectItems.values().toArray(new SelectItem[0]);
+          }
+      } catch (IdUnusedException ex) {
+          log.warn("getGroupsForSite no site found", ex);
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/PublishedAssessmentSettingsBean.java:1488]:
```suggestion
          log.warn("No site found while attempting to get groups, {}", ex.toString());
```

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java [hunk 3]
```diff
@@ -1629,30 +1631,20 @@ public void resetOriginalDateString() {
...
-                  groupSelectItems[i++] = (SelectItem) sortedSelectItems.get(groupIter.next());
+              for (Group group : groups) {
+                  sortedSelectItems.put(group.getTitle(), new SelectItem(group.getId(), group.getTitle()));
               }
+
+              groupSelectItems = sortedSelectItems.values().toArray(new SelectItem[0]);
           }
       } catch (IdUnusedException ex) {
-          // No site available
+          log.warn("getGroupsForSite no site found", ex);
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java:1647]:
```suggestion
          log.warn("No site found while attempting to get groups, {}", ex.toString());
```

**Review by ern — APPROVED** [review]

**Pull request merged**